src/sentry/monitors/consumers/check_in.py

```diff
@@ -17,14 +17,63 @@
     MonitorCheckIn,
     MonitorEnvironment,
     MonitorStatus,
+    MonitorType,
 )
 from sentry.monitors.utils import signal_first_checkin
+from sentry.monitors.validators import CronJobConfigValidator
 from sentry.utils import json
 from sentry.utils.dates import to_datetime
 
 logger = logging.getLogger(__name__)
 
 
+def _ensure_monitor_with_config(
+    project: Project,
+    monitor_slug: str,
+    config: Optional[Dict],
+):
+    try:
+        monitor = Monitor.objects.select_for_update().get(
+            slug=monitor_slug,
+            project_id=project.id,
+            organization_id=project.organization_id,
+        )
+    except Monitor.DoesNotExist:
+        monitor = None
+
+    if not config:
+        return monitor
+
+    validator = CronJobConfigValidator(data=config)
+
+    if not validator.is_valid():
+        logger.debug("monitor_config for %s is not valid", monitor_slug)
+        return monitor
+
+    validated_config = validator.validated_data
+    created = False
+
+    # Create monitor
+    if not monitor:
+        monitor, created = Monitor.objects.update_or_create(
```

Comment:
are we doing `update_or_create` as a safety check instead of just `create`?

Reply:
Correct we're doing this to avoid race conditions